### PR TITLE
send channel name as a custom dimension

### LINKED ISSUE #10696: Request for analytics: clearly send the "channel" our users have installed
Today, we seem to send a very specific "app version" , such as "alpha/48934y3", "master/47fh4h34f", etc.

We were just trying to answer "how many of our users are on the alpha channel" ? We found this difficult, as we had to find all master and alpha versions and then sum them.

Our suggestion: send a "custom dimension" for "channel", and populate it with just the branch name. (Similar to how we send os_version as a custom dimension today, for the host dev machine)

Thanks!

### PR DESCRIPTION
**Pull request by sethladd**
Fixes https://github.com/flutter/flutter/issues/10696

### CODE REVIEW

**Comment by sethladd** [comment]
cc @hixie @devoncarew

File: packages/flutter_tools/lib/src/usage.dart [hunk 1]
```diff
@@ -29,6 +29,9 @@ class Usage {
     // default as custom dimension 1 (configured in our analytics account).
     _analytics.setSessionValue('dimension1', os.name);
 
+    // Report the branch (or, "channel") name.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/usage.dart]: I'd remove this comment (it doesn't really add anything you can't figure out from the code), remove the blank line before it, and then update the earlier comment to cover both cases.

**Comment by devoncarew** [comment]
lgtm w/ Hixie's suggestion

**Comment by sethladd** [comment]
@devoncarew @Hixie ptal thanks!

File: packages/flutter_tools/lib/src/usage.dart [hunk 2]
```diff
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:io';
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/usage.dart]: Is this import used?

Comment (devoncarew) [inline packages/flutter_tools/lib/src/usage.dart]: (I don't believe we generaly want to import dart:io in flutter_tools)

Comment (devoncarew) [inline packages/flutter_tools/lib/src/usage.dart]: ah, which explains this guy: https://travis-ci.org/flutter/flutter/jobs/244783487#L314

**Comment by devoncarew** [comment]
lgtm w/ the removed import

**Comment by sethladd** [comment]
Ah, thanks! That was left over from debugging.

**Pull request merged**